### ISSUE TITLE
BinExport files result in many keyerror issues

### ISSUE THREAD

**Issue by maximus12793** [quarkslab/qbindiff#35]
Following the documentation I've tried `secondary = Program(LoaderType.binexport, "./objcopy.BinExport")`
and `p1 = Program(Path("objcopy.BinExport"))` both of which crash in unexpected ways.

```
...
    91     dst = cg.vertex[edge.target_vertex_index].address
     92     self.callgraph.add_edge(src, dst)
---> 93     self[src].children.add(self[dst])
     94     self[dst].parents.add(self[src])
     96 # Create a map of function names for quick lookup later on
...
  self.be_prog = binexport.ProgramBinExport(file)
  File ".../python3.10/site-packages/binexport/program.py", line 93, in __init__
    self[src].children.add(self[dst])
KeyError: 4751856
```

When using the cli via `qbindiff -l 'binexport' file1.BinExport file2.BinExport` I get the same sort of key errors as well despite these files working fine with BinDiff as-is.

Note: If I do not explicitly ask for `binexport` I get this issue

```
     59     self._backend = ProgramBackendQuokka(*args, **kwargs)
     61 else:
---> 62     raise NotImplementedError("Loader: %s not implemented" % loader)
     64 self._filter = lambda x: True
     65 self._load_functions()
```

Any idea what could be causing the issue? I am using BinDiff 8 and Ghidra 10.3

**Comment by Fenrisfulsur** [comment]
Hello [maximus12793](https://github.com/maximus12793), 

We are aware if this problem, it's because Ghidra creates reference to unknown function that doesn't exist in the exported file. The  problem is actually related to `python-binexport`, We are currently working on a patch. 

If you want to have it working now, you can apply the fix from [here](https://github.com/quarkslab/python-binexport/pull/8) to your local install of `python-binexport`. 

For the second issue, you can try without the `-l` option, it will use BinExport by default.

If you have still some issues, let me know :)

**Comment by maximus12793** [comment]
@Fenrisfulsur thanks for the quick reply! Just to confirm, the documentation of `python-binexport` says it requires IDA. Based on the commit you linked, it looks like this should just help process the file and I am ok to keep using the Ghidra setup I have today? I do not have an IDA instance currently.

```
As python-binexport entirely relies on Binexport, it has to be installed first. The project is available at: https://github.com/google/binexport

Note that python-binexport requires IDA >=7.2 (as it calls the BinExportBinary IDC function).
```

**Comment by Fenrisfulsur** [comment]
No, there is no need to use IDA, you can just use the Ghidra exporter.

**Comment by RobinDavid** [comment]
Thanks @maximus12793 for the feedback. Yes at the time of writing Binexport only had an IDA plugin. I just updated the README to make it more explicit: https://github.com/quarkslab/python-binexport/commit/2fe404cd88db9f2016ad35669ab70269be1ccfe1.

For the issue, yes we are aware of it, it shall get fixed soon.

**Comment by Fenrisfulsur** [comment]
Sorry for the delay, I did not receive the notification. Yeah, this issue tends to happen often when using the Ghidra Exporter. This is mainly due to errors in Ghidra analysis that are not correctly handled by BinExport plugin. We are currently working on a Quokka exporter for Ghidra to address these issues.
